tests/test_commandLine.py

```diff
@@ -0,0 +1,105 @@
+# This file is part of daf_butler.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Unit tests for daf_butler command line interface commands.
+"""
+
+import click
+import click.testing
+import os
+import unittest
+
+from lsst.daf.butler.cli import butler
+
+
+TESTDIR = os.path.abspath(os.path.dirname(__file__))
+
+os.environ["DAF_BUTLER_PLUGINS"] = "lsst.daf.butler.cli.cmd"
+
+
+class ButlerCliTestSuite(unittest.TestCase):
+
+    def testCreate(self):
+        """Test creating a repostiry."""
+        runner = click.testing.CliRunner()
+        with runner.isolated_filesystem():
+            result = runner.invoke(butler.getCli(), ["create", "here"])
+            self.assertEqual(result.exit_code, 0)
+
+    def testCreate_outfile(self):
+        """Test creating a repository and specify an outfile location."""
+        runner = click.testing.CliRunner()
+        with runner.isolated_filesystem():
+            result = runner.invoke(butler.getCli(), ["create", "here", "--outfile=there"])
+            self.assertEqual(result.exit_code, 0)
+            self.assertTrue(os.path.exists("there"))  # verify the separate config file was made
+            with open("there", "r") as f:
+                contents = " ".join(f.readlines())
+                # verify a few expected strings are in the config file:
+                self.assertTrue("datastore" in contents)
+                self.assertTrue("PosixDatastore" in contents)
+                self.assertTrue("root" in contents)
+
+    def testDumpConfig(self):
+        """Test dumping the config to stdout."""
+        runner = click.testing.CliRunner()
+        with runner.isolated_filesystem():
+            result = runner.invoke(butler.getCli(), ["create", "here"])
+            self.assertEqual(result.exit_code, 0)
+
+            # test dumping to stdout:
+            result = runner.invoke(butler.getCli(), ["dump-config", "here"])
+            self.assertEqual(result.exit_code, 0)
+            # check for some expected keywords:
+            self.assertTrue("composites" in result.stdout)
```

Comment:
Hmm. Can't we read the output as YAML, parse that, and then check for `composites` in the `dict`?

Also, please use `assertIn`.